### ISSUE TITLE
Include version element for use in Citation

### ISSUE THREAD

**Issue by n-a-t-e** [cioos-siooc/metadata-entry-form#208]
There's no space for a version number for the data in the CIOOS metadata profile but it could go in MD_DataIdentification.citation.CI_Citation.edition according to ESIP Data Citation Guidelines. This would be useful for versioning datasets using the 'snapshot' approach.

This is how it would look in ISO 19115-3. I noticed theres also an optional `editionDate` field, which must be a datetime.

In `/mdb:MD_Metadata/mdb:identificationInfo/mri:MD_DataIdentification/mri:citation/cit:CI_Citation`:

```xml
...
</cit:date>	

<cit:edition>
    <gco:CharacterString>1.23</gco:CharacterString>
</cit:edition>

<cit:editionDate>
    <gco:DateTime>2020-03-07T02:02:22Z</gco:DateTime>
</cit:editionDate>
```

This refers to versioning of the metadata record:

`edition`  will come from a free form text field in the metadata form

`editionDate` would be last edited date that is already stored in the record

If citation-js supports `version` as an input, we could include the version in the citation as well

**Comment by fostermh** [comment]
here is a link for how to interpret citations in the context of MD_DataIdentification https://wiki.esipfed.org/CI_Citation_for_MD_DataIdentification.

In this case, I think,  we should use one of 'edition' or 'editionDate' but not both, at least according to that link. 

When you mention editionDate and say it will be populated by 'last edited date' are you talking about the date the metadata form was last edited? If so, I'm not sure this is correct as I think the date should relate to the data. for example, you might have a dataset snapshot ending in 2022-12-31 but edit the metadata form in January.

**Comment by BryceMarshall** [comment]
Sent https://github.com/cioos-siooc/metadata-entry-form/pull/223 & https://github.com/cioos-siooc/metadata-xml/pull/106

I've just used the version field there

**Comment by Br-Johnson** [comment]
@n-a-t-e the key here is to assign a version to the data as @fostermh points out. I'm less concerned about versioning the metadata.

**Comment by n-a-t-e** [comment]
So these are the date fields we are using and how we do/will describe them in the metadata form:

Start/end collection date (date) `/mdb:MD_Metadata/mdb:identificationInfo/mri:MD_DataIdentification/mri:extent/gex:EX_Extent/gex:temporalElement/gex:EX_TemporalExtent/gex:extent/gml:TimePeriod`

Dataset edition (free text) `/mdb:MD_Metadata/mdb:identificationInfo/mri:MD_DataIdentification/mri:citation/cit:CI_Citation/cit:edition` **\*new**

Dataset edition date (date) `/mdb:MD_Metadata/mdb:identificationInfo/mri:MD_DataIdentification/mri:citation/cit:CI_Citation/cit:editionDate` **\*new**

It looks like in our ISO profile we could use both `edition` and `editionDate`, Oxygen isn't complaining about having both

I could see users asking what this new Edition Date field is referring to, any ideas for how to explain this? 

Something like

_Enter a version date and/or name_

**Comment by Br-Johnson** [comment]
editionDate seems extraneous and could easily get conflated with revision date. 

I think in the metadata intake form we should call refer to it as 'Dataset version (eg. v1.2)'.

**Comment by Br-Johnson** [comment]
Ideally the version number can appear in the citation as well both in the intake form preview and in CKAN.

**Comment by fostermh** [comment]
regarding citation.js: it appears `version` is the correct key to use when adding version to the citation. It does support edition and edition-date however these fields add suffixes to the number such as 'ed' 

for example using edition=25 results in `(25th ed.) being added to the citation. Wher using version=23 results in `(Version 23)`

**Comment by n-a-t-e** [comment]
We also currently have a "Revision date" being stored at `/mdb:MD_Metadata/mdb:identificationInfo/mri:MD_DataIdentification/mri:citation/cit:CI_Citation/cit:date/cit:CI_Date/cit:date/gco:Date`

**Comment by n-a-t-e** [comment]
Fixed by #223 and https://github.com/cioos-siooc/metadata-xml/pull/106
